### PR TITLE
TF-2078 Add `yes` ,`no` & `maybe` option in email calendar event

### PR DESCRIPTION
**Pull request by dab246**
## Issue

#2078 

## Resolved

1. Display options

https://github.com/linagora/tmail-flutter/assets/80730648/0eec23e4-380b-4069-af52-f5ce4cbc1194

3. Perform action for options

https://github.com/linagora/tmail-flutter/assets/80730648/426be72e-fdfa-4200-a798-ff3247fb19a7

### CODE REVIEW

File: lib/features/email/domain/usecases/parse_calendar_event_interactor.dart [hunk 1]
```diff
@@ -11,11 +13,26 @@ class ParseCalendarEventInteractor {
...
+    String emailContents
+  ) async* {
     try {
       yield Right(ParseCalendarEventLoading());
-      final calendarEventList = await _calendarEventRepository.parse(accountId, blobIds);
-      yield Right(ParseCalendarEventSuccess(calendarEventList));
+
+      final result = await Future.wait(
+        [
+          _calendarEventRepository.parse(accountId, blobIds),
```
Comment (hoangdat) [inline lib/features/email/domain/usecases/parse_calendar_event_interactor.dart:26]: one question, what happen if we can not `parse` the calendar event? How about use the original content?

**Comment by hoangdat** [comment]
- [ ] in blue bar also need these options
image

File: lib/features/email/domain/usecases/parse_calendar_event_interactor.dart (hunk 1 above)
Comment (dab246) [reply lib/features/email/domain/usecases/parse_calendar_event_interactor.dart:26]: If the calendar content cannot be parsed, we will display the original content of the email (like the old versions).

**Comment by chibenwa** [comment]
We can use `keywords` on the email to remember the decision: yes-no-maybe

**Comment by chibenwa** [comment]
And how is the event answered? 
Send an ics by email updating the attendance?

**Comment by dab246** [comment]
> We can use `keywords` on the email to remember the decision: yes-no-maybe

I don't understand what is the `keywords` value for decision `yes, no, maybe`. I read the jmap documentation but didn't find this.

**Comment by hoangdat** [comment]
can not understand your comment too.

**Comment by quantranhong1999** [comment]
> We can use keywords on the email to remember the decision: yes-no-maybe

I guess Benoit suggests labeling the email (with event invitation) by JMAP keywords, then users can filter which event they accept/reject/maybe.

File: lib/features/email/presentation/widgets/email_attachments_widget.dart [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+  @override
+  State<EmailAttachmentsWidget> createState() => _EmailAttachmentsWidgetState();
+}
+
+class _EmailAttachmentsWidgetState extends State<EmailAttachmentsWidget> {
+
+  final _imagePaths = Get.find<ImagePaths>();
+
+  bool _isDisplayAll = false;
```
Comment (sherlockvn) [inline lib/features/email/presentation/widgets/email_attachments_widget.dart:36]: you can try `ValueListenerBuilder`

File: lib/features/email/presentation/widgets/email_receiver_widget.dart [hunk 3]
```diff
@@ -15,59 +15,67 @@ import 'package:model/extensions/list_email_address_extension.dart';
...
+        if (_isDisplayAll)
           Padding(
             padding: EdgeInsets.symmetric(
               vertical: DirectionUtils.isDirectionRTLByLanguage(context) ? 0 : 6),
             child: MaterialTextButton(
               padding: DirectionUtils.isDirectionRTLByLanguage(context)
                 ? const EdgeInsets.symmetric(horizontal: 8, vertical: 4)
                 : null,
-              onTap: controller.collapseEmailAddress,
+              onTap: () => setState(() => _isDisplayAll = false),
```
Comment (sherlockvn) [inline lib/features/email/presentation/widgets/email_receiver_widget.dart:73]: same

File: lib/features/email/presentation/widgets/email_attachments_widget.dart (hunk 2 above)
Comment (dab246) [reply lib/features/email/presentation/widgets/email_attachments_widget.dart:36]: The variable `_isDisplayAll ` is used in many sub-widgets of `EmailAttachmentWidget` so we should not use `ValueListenerBuilder ` here.

**Comment by chibenwa** [comment]
Yes.

Furthermore please detail which actions is triggered by each button.

**Comment by dab246** [comment]
> Yes.
> 
> Furthermore please detail which actions is triggered by each button.
> 
> Are actions implemented or is this just display?

No, currently these buttons are all working but the ones in the original body of the email when clicked. We've filtered and tagged them. You can watch the demo video for better understanding.

Screenshot 2023-08-14 at 15 09 16

**Review by hoangdat — APPROVED** [review]

**Pull request merged**